providers/src/airflow/providers/elasticsearch/log/es_task_handler.py

```diff
@@ -266,17 +275,32 @@ def _render_log_id(self, ti: TaskInstance | TaskInstanceKey, try_number: int) ->
                 data_interval_end = data_interval[1].isoformat()
             else:
                 data_interval_end = ""
-            execution_date = dag_run.execution_date.isoformat()
-
-        return log_id_template.format(
-            dag_id=ti.dag_id,
-            task_id=ti.task_id,
-            run_id=getattr(ti, "run_id", ""),
-            data_interval_start=data_interval_start,
-            data_interval_end=data_interval_end,
-            execution_date=execution_date,
-            try_number=try_number,
-            map_index=getattr(ti, "map_index", ""),
+            logical_date = (
+                dag_run.logical_date.isoformat() if AIRFLOW_V_3_0_PLUS else dag_run.execution_date.isoformat()
+            )
+
+        return (
+            log_id_template.format(
+                dag_id=ti.dag_id,
+                task_id=ti.task_id,
+                run_id=getattr(ti, "run_id", ""),
+                data_interval_start=data_interval_start,
+                data_interval_end=data_interval_end,
+                logical_date=logical_date,
+                try_number=try_number,
+                map_index=getattr(ti, "map_index", ""),
+            )
+            if AIRFLOW_V_3_0_PLUS
+            else log_id_template.format(
+                dag_id=ti.dag_id,
+                task_id=ti.task_id,
+                run_id=getattr(ti, "run_id", ""),
+                data_interval_start=data_interval_start,
+                data_interval_end=data_interval_end,
+                execution_date=logical_date,
+                try_number=try_number,
+                map_index=getattr(ti, "map_index", ""),
+            )
         )
```

Comment:
```suggestion
        log_id_template.format(
            dag_id=ti.dag_id,
            task_id=ti.task_id,
            run_id=getattr(ti, "run_id", ""),
            data_interval_start=data_interval_start,
            data_interval_end=data_interval_end,
            execution_date=logical_date,
            logical_date=logical_date,
            try_number=try_number,
            map_index=getattr(ti, "map_index", ""),
        )
```

`str.format` ignores extra arguments, so we can just add both keys with checking version.

Reply:
Changed it